### DISCUSSION TITLE
Regression testing against `mozjpeg` 4.1.3 fails

### DISCUSSION THREAD

**Discussion by p-linnane** [libvips/libvips#3508] [General]
## Bug report

**Describe the bug**
Hello 👋 . I'm a maintainer for the [Homebrew](https://brew.sh) project. While doing some regression testing for the new mozjpeg release, we encountered some failures with vips on macOS 12-x86_64 and macOS 11-x86_64.

**To Reproduce**
1. Run Homebrew package testing for vips against mozjpeg 4.1.3
https://github.com/Homebrew/homebrew-core/actions/runs/5061915590/jobs/9089098739

**Expected behavior**
Test passes

**Actual behavior**
```
  ==> /usr/local/Cellar/vips/8.14.2_1/bin/vips jpegsave /usr/local/Homebrew/Library/Homebrew/test/support/fixtures/test.png /private/tmp/vips-test-20230523-47716-1c22jyq/test.jpg --trellis-quant 2>&1
  sh: line 1: 47925 Illegal instruction: 4  /usr/local/Cellar/vips/8.14.2_1/bin/vips jpegsave /usr/local/Homebrew/Library/Homebrew/test/support/fixtures/test.png /private/tmp/vips-test-20230523-47716-1c22jyq/test.jpg --trellis-quant 2>&1
```

**Screenshots**
N/A

**Environment**
(please complete the following information)
 - OS: macOS 11 (Intel), macOS 12 (Intel)
 - Vips: 8.14.2

**Additional context**
Relates to https://github.com/Homebrew/homebrew-core/pull/131797

**Comment by lovell** [comment]
Hi Patrick, thanks for all your work helping to maintain Homebrew!

This is sadly a problem in mozjpeg v4.1.3, which has merged libjpeg-turbo v2.1.5 but not the latest v2.1.5.1

https://github.com/libjpeg-turbo/libjpeg-turbo/issues/649
https://github.com/mozilla/mozjpeg/pull/432

Possible options for Homebrew (there may be others):

* Pin mozjpeg to v4.1.1 for now
* Apply the patch from https://github.com/mozilla/mozjpeg/pull/432 as part of the mozjpeg v4.1.3 upgrade

**Reply by p-linnane** [reply]
Confirming this patch fixed the issue for us. Thanks for your quick help!
